app/controllers/application_controller.rb

```diff
@@ -650,14 +651,14 @@ def build_ae_tree(type = :ae, name = :ae_tree)
 
   # moved this method here so it can be accessed from pxe_server controller as well
   def log_depot_set_verify_status
-    if (@edit[:new][:log_password] == @edit[:new][:log_verify]) && @edit[:new][:uri_prefix] != "nfs" &&
```

Comment:
@mzazrivec I would use extra variables to improve readability here:

```diff
- @edit[:log_verify_status] = if (@edit[:new][:log_password] == @edit[:new][:log_verify]) && @edit[:new][:uri_prefix] != "nfs" &&
-                                (@edit[:new][:uri].present? && @edit[:new][:log_userid].present? && @edit[:new][:log_password].present? && @edit[:new][:log_verify].present?)
-                               true
-                             elsif @edit[:new][:uri_prefix] == "nfs" && @edit[:new][:uri].present?
-                               true
-                             else
-                               false
-                             end
+ uri_filled = @edit[:new][:uri].present?
+ nfs_prefix = (@edit[:new][:uri_prefix] == "nfs")
+ passwords_match = (@edit[:new][:log_password] == @edit[:new][:log_verify])
+ authentication_credentials_filled = (
+  uri_filled &&
+  @edit[:new][:log_userid].present? &&
+  @edit[:new][:log_password].present? &&
+  @edit[:new][:log_verify].present?
+ )
+
+ @edit[:log_verify_status] = (
+   (nfs_prefix && uri_filled) ||
+   (!nfs_prefix && authentication_credentials_filled && passwords_match)
+ )
```

...what do you think?

Reply:
I simplified the condition inside that routine and also moved the routine to `PxeController`, since it's used only in that controller.